### ISSUE TITLE
cross_entropy_loss edge case

### ISSUE THREAD

**Issue by DanTaranis** [pytorch/pytorch#72056]
### 🐛 Describe the bug

I tried to use label smoothing in cross enthropy and I guess it hit some edge case that fails:

```
import torch
input = torch.FloatTensor([[ 0.1735,  0.2191],
        [-0.1377,  0.3467],
        [-0.2162,  0.2021],
        [-0.1177,  0.2157],
        [-0.1232,  0.2620],
        [-0.1578,  0.4038],
        [-0.0209,  0.3298],
        [-0.0639,  0.2317],
        [-0.3914,  0.2106],
        [-0.4211,  0.3420]])
target =torch.LongTensor([ 0, -1,  0,  0,  0,  0,  0,  0, -1,  0])
weight = torch.FloatTensor([1,16])
reduction = 'mean'
ignore_index = -1
label_smoothing = .1
torch._C._nn.cross_entropy_loss(input, target, weight, _Reduction.get_enum(reduction), ignore_index, label_smoothing)
```
results:
```
RuntimeError: index -1 is out of bounds for dimension 0 with size 2
```

### Versions

torch version : '1.10.2+cu113'

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345

**Comment by jbschlosser** [comment]
Hey @DanTaranis, thanks for the report! This should be fixed on master, thanks to #69511. Please feel free to reopen this issue if you still encounter any problems.

**Comment by danFromTelAviv** [comment]
excellent thanks
